skbase/tests/test_base.py

```diff
@@ -966,6 +966,45 @@ class TestClass(BaseObject):
         assert "foo" not in test_obj_clone.get_config().keys()
 
 
+@pytest.mark.parametrize("clone_config", [True, False])
+def test_nested_config_after_clone_tags(clone_config):
+    """Test clone also clones config of nested objects."""
+
+    class TestClass(BaseObject):
+        _tags = {"some_tag": True, "another_tag": 37}
+        _config = {"check_clone": 0}
+
+        def __init__(self, nested=None):
+            self.nested = nested
+
+    test_obj = TestClass(
+        nested=[TestClass().set_config(**{"check_clone": 42, "foo": "bar"})]
```

Comment:
we should also check `nested` being another estimator (not wrapped in a list) - I think that would fail with th current fix, see above.

Reply:
Good point! I updated the test to also include an estimator parameter. I think that test now covers the behaviour we need to cover here however I would like to refactor it as the implementation is quite _ugly_ currently. Maybe this could be a different PR where we clean-up/refactor the `test_config_after_clone_tags` and `test_nested_config_after_clone_tags` tests

Reply:
tests don't need to win beauty contests, they need to be correct, easy to understand, and cover relevant cases, all of which I belive is true even right now.

However, I see where you are coming from, so by all means!